### PR TITLE
Tester integration, Noise2Score, python version 3.9

### PR DESCRIPTION
**Pull request by tachella**
- homogeinizing python versions to 3.9+
- small fixes to gaussian noise to use with SigmaGenerator
- Added Gamma Noise
- Added Score Loss (there was one existing version but it was hidden from the API and didn't work)
- Moved all testing to Trainer
- Added no-learning reconstruction to Trainer.
- SURE improvements
- Added ideal sinc filter with 'optimal' kaiser windowing.
- Added windows machine in github actions testing 
- fix wandb logging issues

### Checks to be done before submitting your PR
- [x] `python3 -m pytest tests/` runs successfully.
- [x] `black .` runs successfully.
- [x] `make html` runs successfully (in the `docs/` directory).
- [x] Updated docstrings related to the changes (as applicable).
- [ ] Added an entry to the [CHANGELOG.rst](../CHANGELOG.rst).

### CODE REVIEW

File: deepinv/loss/measplit.py [hunk 1]
```diff
@@ -108,6 +108,7 @@ def forward(self, x_net, y, physics, model, **kwargs):
         """
         mask = model.get_mask()
 
+        mask *= getattr(physics, "mask", 1.0)
```
Comment (Andrewwango) [inline deepinv/loss/measplit.py:111]: What is the reasoning here? If a `physics.mask` is present (e.g. in MRI), `mask` will already be subsampled from `physics.mask` (in the `SplittingModel.forward`), so this line does nothing.

Comment (tachella) [reply deepinv/loss/measplit.py:111]: it is a safety line for the case where the second mask is not subsampled.

Comment (Andrewwango) [reply deepinv/loss/measplit.py:111]: I have added this in #279 now.

File: deepinv/loss/score.py [hunk 2]
```diff
@@ -1,41 +1,161 @@
...
 from deepinv.loss.loss import Loss
 
 
 class ScoreLoss(Loss):
     r"""
-    Approximates score of a distribution.
+    Learns score of noise distribution.
 
-    It approximates the score of the measurement distribution :math:`S(y)\approx \nabla \log p(y)`
+    Approximates the score of the measurement distribution :math:`S(y)\approx \nabla \log p(y)`
```
Comment (samuro95) [inline deepinv/loss/score.py:10]: isn't it p_\sigma rather than p ?

File: deepinv/loss/score.py [hunk 3]
```diff
@@ -1,41 +1,161 @@
...
     https://proceedings.neurips.cc/paper_files/paper/2021/file/077b83af57538aa183971a2fe0971ec1-Paper.pdf.
 
     The score loss is defined as
 
     .. math::
 
-        \| \epsilon - \sigma S(y+ \sigma \epsilon) \|^2
+        \| \epsilon + \sigma S(y+ \sigma \epsilon) \|^2
```
Comment (samuro95) [inline deepinv/loss/score.py:17]: This is the same loss for all noise models ? or is it only valid for Gaussian noise ?

File: deepinv/loss/score.py [hunk 4]
```diff
@@ -1,41 +1,161 @@
...
+        :param torch.nn.Module model: Backbone model approximating the score.
+        :return: (torch.nn.Module) Adapted reconstruction model.
+        """
+        if isinstance(model, ScoreModel):
+            return model
+        else:
+            return ScoreModel(model, self.noise_model, self.delta, self.total_batches)
+
+
+class ScoreModel(torch.nn.Module):
```
Comment (samuro95) [inline deepinv/loss/score.py:106]: Missing some docstring for the arguments

File: deepinv/loss/score.py (hunk 2 above)
Comment (tachella) [reply deepinv/loss/score.py:10]: yes, but they anneal sigma to zero during training.

File: deepinv/loss/score.py [hunk 5]
```diff
@@ -1,41 +1,161 @@
...
-        :return: (torch.Tensor) loss.
+        :param torch.Tensor y: Measurements.
+        :param deepinv.physics.Physics physics: Forward operator associated with the measurements.
+        :param torch.nn.Module model: Reconstruction model.
+        :return: (torch.Tensor) Score loss.
         """
-        std = np.randn() * self.delta
-        noise = torch.randn_like(y)
-        return self.metric(noise, std * model(std * noise + y))
+        return model.get_error()
```
Comment (samuro95) [inline deepinv/loss/score.py:91]: Can we add a test to ensure that we indeed computer the error beforehand ?

Comment (tachella) [reply deepinv/loss/score.py:91]: do you mean a pytest? or a flag that makes sure that the error has been recomputed?

File: deepinv/loss/score.py (hunk 4 above)
Comment (tachella) [reply deepinv/loss/score.py:106]: now it should be fine right?
